File: pyg-parent/pyg-manager-web/src/main/webapp/js/service/brandService.js
app.service('brandService',function ($http) {
    this.findAll = function () {
        return $http.get("../brand/findAll.do");
    }

    this.findByPage = function (page,size) {
        return $http.get("../brand/findByPage.do?page= " + page + "&size=" + size);
    }

    this.findOne = function (id) {
        return $http.get("../brand/findOne.do?id=" + id);
    }

    this.add = function (newEntity) {
        return $http.post("../brand/add.do", newEntity);
    }

    this.update = function (newEntity) {
        return $http.post("../brand/update.do", newEntity);
    }

    this.del = function (selectIds) {
        return $http.get("../brand/delete.do?ids="+ selectIds);
    }

    this.search = function (page,size,searchEntity) {
        return $http.post("../brand/search.do?page="+ page + "&size=" + size,searchEntity);
    }

    this.selectOptionList=function () {
        return $http.get('../brand/selectOptionList.do');
    }
})

File: pyg-parent/pyg-sellergoods-interface/src/main/java/com/pyg/service/BrandService.java
package com.pyg.service;

import com.pyg.pojo.TbBrand;
import entity.PageResult;

import java.util.List;
import java.util.Map;


public interface BrandService {

    /**
     * 查询所有brand
     * @return
     */
    public List<TbBrand> findAll();


    /**
     * 品牌分页
     * @param pageNum  当前页码
     * @param pageSize 每页记录数
     * @return
     */
    public PageResult findByPage(int pageNum, int pageSize);

    /**
     * 增加
     * @param brand
     */
    public void add(TbBrand brand);

    /**
     * 根据id查询
     * @param id
     * @return
     */
    public TbBrand findOne(Long id);

    /**
     * 更新brand
     * @param brand
     */
    public void update(TbBrand brand);

    /**
     * 删除选中
     * @param ids
     */
    public void delete(Long[] ids);

    /**
     * 品牌分页
     * @param brand
     * @param pageNum
     * @param pageSize
     * @return
     */                             //将模糊条件都封装到实体类 brand 中
    public PageResult findByPageAndCondition(TbBrand brand,int pageNum ,int pageSize);

    /**
     * 返回下拉列表数据
     * @return
     */
    public List<Map> selectOptionList();

}
